sound/soc/codecs/rt711-sdw.c

```diff
@@ -379,15 +380,15 @@ static int rt711_read_prop(struct sdw_slave *slave)
 	if (!prop->sink_dpn_prop)
 		return -ENOMEM;
 
-	i = 0;
+	j = 0;
```

Comment:
this one is a false positive, `i` is clearly not needed any more, I wouldn't do this. I would just leave this as is, but if we're really keen on pleasing cppcheck, we could initialise `i = -1;` and then `i++`; in the beginning of the loop, then it should be clear, that `i` *is* used after assignment, but I wouldn't bother frankly.

Reply:
Yes it's a false positive but cppcheck found quite a few errors in our code, and it helps to reduce the number of warnings.

this is how it's been fixed for other drivers, so I just added the same recipe for consistency.

Reply:
/me really dislikes making the code less natural for humans to please "intelligent" parsers :-/

Reply:
This is a bit iffy, but we do the same sometimes for performance, so ok for me.